src/main/java/org/graylog/integrations/pagerduty/client/PagerDutyClient.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog.integrations.pagerduty.client;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.annotations.VisibleForTesting;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import org.graylog.events.notifications.EventNotificationContext;
+import org.graylog.integrations.pagerduty.dto.PagerDutyResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.util.Locale;
+
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+
+/**
+ * The Pager Duty REST client implementation class compatible with events V2. For more information
+ * about the event structure please see
+ * <a href="https://v2.developer.pagerduty.com/v2/docs/send-an-event-events-api-v2">the api</a>.
+ *
+ * This class is heavily based on the work committed by Jochen, James, Dennis, Padma, and Edgar
+ * <a href="https://github.com/graylog-labs/graylog-plugin-pagerduty/">here</a>.
+ *
+ * @author Jochen Schalanda
+ * @author James Carr
+ * @author Dennis Oelkers
+ * @author Padma Liyanage
+ * @author Edgar Molina
+ */
+public class PagerDutyClient {
+    private static final Logger LOG = LoggerFactory.getLogger(PagerDutyClient.class);
+
+    @VisibleForTesting
+    static final String API_URL = "https://events.pagerduty.com/v2/enqueue";
+
+    private final OkHttpClient httpClient;
+    private final ObjectMapper objectMapper;
+    private final MessageFactory messageFactory;
+
+    @Inject
+    public PagerDutyClient(final OkHttpClient httpClient,
+                    final ObjectMapper objectMapper,
+                    final MessageFactory messageFactory) {
+        this.httpClient = httpClient;
+        this.objectMapper = objectMapper;
+        this.messageFactory = messageFactory;
+    }
+
+    public PagerDutyResponse trigger(EventNotificationContext ctx) throws PagerDutyClientException {
```

Comment:
@waab76 - shouldn't this use `TemporaryEventNotificationException` and `PermanentEventNotificationException`

Reply:
I didn't want to use notification-specific exceptions in the PagerDutyClient because it could be used for other things in the future.  There is no reason to directly link the PagerDutyClient to our Event Notification system by using Event Notification exceptions here.